starknet-providers/src/provider.rs

```diff
@@ -298,6 +301,21 @@ pub trait Provider {
             Err(ProviderError::ArrayLengthMismatch)
         }
     }
+
+    async fn batch_requests<I, P>(
+        &self,
+        requests: I,
+    ) -> Result<Vec<serde_json::Value>, ProviderError>
```

Comment:
Do we have to resort to `serde_json::Value` here? I generally dislike having that as part of the interface.

Reply:
Good question. Not necessarily, however this function was specifically introduced for this approach - to return generic JSON type based on the type returned. Otherwise, we'd have to create custom types for this and that would limit the ways we can call it

It's possible to change it if you prefer any other direction

Reply:
ty @tcoratger , @xJonathanLEI can you please let us know your thoughts? ty